src/search/search.component.ts

```diff
@@ -0,0 +1,236 @@
+				<button
+					class="bx--search-close"
+					[ngClass]="{
+						'bx--search-close--hidden': !value || value === 0
+					}"
+					title="Clear search input"
+					[attr.aria-label]="ariaLabel"
+					(click)="clearSearch()">
+					<svg
+						width="16"
+						height="16"
+						viewBox="0 0 16 16"
+						xmlns="http://www.w3.org/2000/svg">
+						<path
+							d="M8 6.586L5.879 4.464 4.464 5.88 6.586 8l-2.122 2.121 1.415 1.415L8 9.414l2.121 2.122 1.415-1.415L9.414
+							   8l2.122-2.121-1.415-1.415L8 6.586zM8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16z"
+							fill-rule="evenodd"/>
+					</svg>
+				</button>
+			</div>
+		</div>
+	`,
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: Search,
+			multi: true
+		}
+	]
+})
+export class Search implements ControlValueAccessor {
+	/**
+	 * Variable used for creating unique ids for search components.
+	 */
+	static searchCount = 0;
+
+	/**
+	 * `light` or `dark` search theme.
+	 */
+	@Input() theme: "light" | "dark" = "dark";
+	/**
+	 * Size of the search field.
+	 */
+	@Input() size: "sm" | "lg" = "lg";
+	/**
+	 * Set to `true` for a disabled search input.
+	 */
+	@Input() disabled = false;
+	/**
+	 * Sets the name attribute on the `input` element.
+	 */
+	@Input() name: string;
+	/**
+	 * The unique id for the search component.
+	 */
+	@Input() id = `search-${Search.searchCount}`;
+	/**
+	 * Reflects the required attribute of the `input` element.
+	 */
+	@Input() required: boolean;
+	/**
+	 * Sets the value attribute on the `input` element.
+	 */
+	@Input() value = "";
+	/**
+	 * Sets the placeholder attribute on the  `input` element.
+	 */
+	@Input() placeholder = "Search";
+	/**
+	 * Used to set the `aria-label` attribute on the input element.
+	 */
+	// tslint:disable-next-line:no-input-rename
+	@Input("aria-label") ariaLabel = "Clear search input";
+	/**
+	 * Used to set the `aria-labelledby` attribute on the input element.
+	 */
+	// tslint:disable-next-line:no-input-rename
+	@Input("aria-labelledby") ariaLabelledby: string;
+	/**
+	 * Emits event notifying other classes when a change in state occurs in the input.
+	 */
+	@Output() change = new EventEmitter<SearchChange>();
+
+	/**
+	 * Creates an instance of `Search`.
+	 * @memberof Search
+	 */
+	constructor() {
+		Search.searchCount++;
+	}
+
+	// this is the initial value set to the component
+	public writeValue(value: any) {
+		this.value = value;
+	}
+
+	/**
+	 * Sets a method in order to propagate changes back to the form.
+	 * @param {any} fn
+	 * @memberof Search
+	 */
+	public registerOnChange(fn: any) {
+		this.propagateChange = fn;
+	}
+
+	/**
+	 * Registers a callback to be triggered when the control has been touched.
+	 * @param fn Callback to be triggered when the search input is touched.
+	 */
+	public registerOnTouched(fn: any) {
+		this.onTouched = fn;
+	}
+
+	/**
+	 * Executes on the event of a change within `Search` to block propagation.
+	 * @param {any} event
+	 * @memberof Search
+	 */
+	onChange(event) {
```

Comment:
I'm curious if this is something we need? It's not immediately clear to me why we would need to stop propagation for `change`.

Reply:
Looking at other components (checkbox), I think the idea was to stop the native event propagation and to override it by emitting our own `[ComponentName]Change` object. However, I don't get the point of this behaviour and I agree to stick to the standard.